lib/sync.js

```diff
@@ -164,15 +177,16 @@ module.exports = function (x, options) {
     }
 
     function loadNodeModulesSync(x, start) {
-        var dirs = nodeModulesPaths(start, opts, x);
+        var dirs = packageIterator(x, start, function () {
+            return getPackageCandidates(x, start, opts);
+        }, opts);
+
         for (var i = 0; i < dirs.length; i++) {
             var dir = dirs[i];
-            if (isDirectory(dir)) {
```

Comment:
why is this line removed? if you want to pass your own isDirectory that's fine, but this check still needs to be ran.

Reply:
Maybe it could be moved in the default `getPackageCandidates`?

Reply:
Why is the `isDirectory` needed? Is it an optimization? If `dir` isn't a directory then both `loadAsFile` and `loadAsDirectory` will fail at the next `stat` call, so the loop will be roughly the same 🤔

Reply:
Just like there's a custom `isFile` in v1, there's also a custom `isDirectory` in v2; users are able to overload this logic. If we don't call it, they're unable to do that.